### DISCUSSION TITLE
Working with compressed image HDUs

### DISCUSSION THREAD

**Discussion by JonCook** [nom-tam-fits/nom-tam-fits#277] [Q&A]
Hello,

This is more of a question than an issue.

> In astropy, the header of a compressed image HDU appears to the user like any image header. The actual header stored in the FITS file is that of a binary table HDU with a set of special keywords, defined by the convention, to describe the structure of the compressed image (Z*). The conversion between binary table HDU header and image HDU header is all performed behind the scenes. Since the HDU is actually a binary table, it may not appear as a primary HDU in a FITS file.

See: https://docs.astropy.org/en/stable/io/fits/usage/unfamiliar.html#astropy-io-fits-compressedimagedata

My question is, does `nom.tam.fits` have anything like this? I've seen the `CompressedImageHdu` class but it seems no internal translation is done and if I really want to get the Z* header values I have to ask for them explicitly? 

Many thanks
Jonathan

**Comment by attipaci** [comment]
Hi Jonathan,

I don't have a quick answer. Compression was something Ritchie was working on. I'm not very familiar with it. I will look into it to see what I can make of the code and related tests. You might also want to look at some of the test code. I know there are tests for getting for example BITPIX values from compressed headers. The way the test code is doing it might answer your question...

**Comment by bogdanni** [comment]
@JonCook It has to be done explicitly, it's not very handy. See [here](https://github.com/Helioviewer-Project/JHelioviewer-SWHV/blob/e701211c97f80b541ff4160045745f841f4ac8ba/src/org/helioviewer/jhv/view/uri/FITSImage.java#L55) how it is done in JHV.

**Comment by attipaci** [comment]
It sounds like it could be improved. Any volunteers? :-D

**Comment by JonCook** [comment]
Many thanks @bogdanni
